### PR TITLE
gstreamer: update dependencies

### PR DESCRIPTION
**Pull request by cho-m**
- [ ] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [ ] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [ ] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

Based on https://github.com/Homebrew/homebrew-core/actions/runs/10155278856

### CODE REVIEW

File: Formula/g/gstreamer.rb [hunk 1]
```diff
@@ -52,66 +52,92 @@ class Gstreamer < Formula
 
   depends_on "bison" => :build
   depends_on "cargo-c" => :build
-  depends_on "gitlint" => :build
```
Comment (cho-m) [inline Formula/g/gstreamer.rb:55]: Removed as looks CI-specific - https://gitlab.freedesktop.org/gstreamer/gstreamer/-/commit/5586511446a9988af7b5e19b61c151b5d1111691

File: Formula/g/gstreamer.rb [hunk 2]
```diff
@@ -52,66 +52,92 @@ class Gstreamer < Formula
...
   depends_on "dav1d"
   depends_on "faac"
   depends_on "faad2"
   depends_on "fdk-aac"
   depends_on "ffmpeg@6"
   depends_on "flac"
-  depends_on "gettext"
+  depends_on "gdk-pixbuf"
   depends_on "glib"
-  depends_on "glib-networking"
```
Comment (cho-m) [inline Formula/g/gstreamer.rb:72]: Didn't see any linkage and looks specific to `glib` subproject - https://gitlab.freedesktop.org/gstreamer/gstreamer/-/commit/237f5cc4ee8a20a51d8191e5c96c2b0efa268b5c

File: Formula/g/gstreamer.rb [hunk 3]
```diff
@@ -52,66 +52,92 @@ class Gstreamer < Formula
...
+  depends_on "opencore-amr"
   depends_on "openexr"
+  depends_on "openjpeg"
   depends_on "openssl@3"
   depends_on "opus"
   depends_on "orc"
   depends_on "pango"
   depends_on "pygobject3"
   depends_on "python@3.12"
-  depends_on "rav1e"
```
Comment (cho-m) [inline Formula/g/gstreamer.rb:95]: Didn't see linkage and doesn't seem possible to use library as `rav1e` is a Cargo dependency - https://gitlab.freedesktop.org/gstreamer/gst-plugins-rs/-/blob/main/video/rav1e/Cargo.toml#L14

File: Formula/g/gstreamer.rb [hunk 4]
```diff
@@ -52,66 +52,92 @@ class Gstreamer < Formula
...
+  depends_on "mpg123"
+  depends_on "nettle"
+  depends_on "opencore-amr"
   depends_on "openexr"
+  depends_on "openjpeg"
   depends_on "openssl@3"
   depends_on "opus"
   depends_on "orc"
   depends_on "pango"
   depends_on "pygobject3"
```
Comment (chenrui333) [inline Formula/g/gstreamer.rb:105]:
```
Dependencies with no linkage:
  pygobject3
```

Comment (cho-m) [reply Formula/g/gstreamer.rb:105]: That's because it is Python code for the bindings. Linkage test won't work.

**Review by branchv — APPROVED** [review]

**Pull request merged**